### PR TITLE
Add "How to check downloaded file"

### PR DESCRIPTION
**Pull request by DIRECTcut**
Part of #2 
"How to check downloaded file name"
"Check downloaded file content"

### CODE REVIEW

File: examples/check-downloaded-file-name-and-content/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+fixture `Download file`
+    .page('./index.html')
+    .requestHooks(logger);
+
+test('Check file name', async t => {
+    const fileNameRegEx = /attachment; filename=.*.txt/
+    
+    await t
+        .click('#download-btn')
+        .expect(logger.contains( r => r.response.statusCode === 200 &&
```
Comment (miherlosev) [inline examples/check-downloaded-file-name-and-content/index.js:20]:
```js
.expect(logger.contains(r => {
      if (r.response.statusCode !== 200)
          return false;
     
      const requestInfo = logger.requests[0];
     
      if (!requestInfo)
          return false;

      const downloadedFileName = requestInfo.response.headers['content-disposition'];
      
      if (!downloadedFileName)
          false;

       if (!fileNameRegEx.test(downloadedFileName))
            return false;
      
       const downloadedFileContent = logger.requests[0].response.body;

       return downloadedFileContent  === 'Test content';
}).ok();
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-downloaded-file-name-and-content/index.js [hunk 2]
```diff
@@ -13,10 +13,14 @@ fixture `Download file`
     .requestHooks(logger);
 
 test('Check file name', async t => {
-
+    const fileNameRegEx = /attachment; filename=.*.txt/
```
Comment (miherlosev) [inline examples/check-downloaded-file-name-and-content/index.js:16]: Add a semicolon:

`const fileNameRegEx = /attachment; filename=.*.txt/;`

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Check the Name and the Contents of a Downloaded File
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to obtain the name and the content of a file downloaded during a test.
+
+During the test, a link click starts a file download. To verify download success and obtain the name and the content of the file, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
+
+To verify success of a download, a response status code is checked with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method.
```
Comment (miherlosev) [inline examples/check-downloaded-file-name-and-content/README.MD:9]: Not only status code. File name and file content are checked also.

What about this: 

To verify the downloaded file information the [RequestLog.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method is used in combination with [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method.

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+The Request Logger has the following [options](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/constructor.html) enabled:
+
+|Option|Effect|
+|-|-|
+|`logResponseHeaders`| logs the response header|
+|`logResponseBody`| logs the response body|
+|`stringifyResponseBody`| converts the response body to a string|
+
+The file name is read from the `content-disposition` response header and is logged to the console. The file contents constitute the response body and are logged to the console.
```
Comment (miherlosev) [inline examples/check-downloaded-file-name-and-content/README.MD:21]: I guess it isn't necessary. To see these logs customer should know about opening the browser devtools.

File: examples/check-downloaded-file-name-and-content/index.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Verify file's contents</title>
```
Comment (miherlosev) [inline examples/check-downloaded-file-name-and-content/index.html:5]: `Verify the downloaded file information`

File: examples/check-downloaded-file-name-and-content/index.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { RequestLogger } from 'testcafe';
+
+const url = 'http://localhost:3000/download-file';
+
+const logger = RequestLogger({ url, method: 'GET' }, {
```
Comment (miherlosev) [inline examples/check-downloaded-file-name-and-content/index.js:5]: Align the option values.

```js
const logger = RequestLogger({ url, method: 'GET' }, {
   logResponseHeaders:    true,
   logResponseBody:       true,
   stringifyResponseBody: true
});
```

File: server/index.js [hunk 7]
```diff
@@ -8,12 +8,57 @@ http
...
 
         if (req.url === '/download-file') {
-            const fileStream = fs.createReadStream('./server/data/text-file.txt');
+            const fileStream = fs.createReadStream('./data/text-file.txt');
 
             res.setHeader('content-disposition', 'attachment; filename=text-file.txt');
             fileStream.pipe(res);
         }
+
+        else if (req.url == '/download-random-file') {
```
Comment (miherlosev) [inline server/index.js:17]: Remove the random file generation.

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 7 above)
Comment (DIRECTcut) [reply server/index.js:17]: removed

File: examples/check-downloaded-file-name-and-content/index.js (hunk 2 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/index.js:16]: added

File: examples/check-downloaded-file-name-and-content/index.js (hunk 6 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/index.js:5]: Aligned

File: examples/check-downloaded-file-name-and-content/index.html (hunk 5 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/index.html:5]: edited

File: examples/check-downloaded-file-name-and-content/README.MD (hunk 4 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:21]: > The file name is read from the `content-disposition` response header. The file contents constitute the response body.

+removed `.log`s from the test

File: examples/check-downloaded-file-name-and-content/README.MD (hunk 3 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:9]: > The `RequestLog.contains` method in combination with `t.expect.ok` method verify the downloaded file information.

File: examples/check-downloaded-file-name-and-content/index.js (hunk 1 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/index.js]: edited

File: server/index.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
 const http = require('http');
-const fs   = require('fs');
+const fs = require('fs');
```
Comment (miherlosev) [inline server/index.js:2]: Lose align
```js
const http = require('http');
const fs   = require('fs');
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 8 above)
Comment (DIRECTcut) [reply server/index.js:2]: strange, I have it perfectly aligned
image

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# Check the Name and the Contents of a Downloaded File
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to obtain the name and the content of a file downloaded during a test.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:5]: edited

File: server/index.js (hunk 8 above)
Comment (DIRECTcut) [reply server/index.js:2]: edited

**Review by miherlosev — APPROVED** [review]

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# Check the Name and the Contents of a Downloaded File
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to check the name and the content of a file downloaded during a test.
+
+During the test, a link click starts a file download. To verify download success and obtain the name and the content of the file, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
```
Comment (VasilyStrelyaev) [inline examples/check-downloaded-file-name-and-content/README.MD:7]: > a link click starts a file download

`a link click starts a file download` sounds like "a cannon shot starts the festival". Maybe it's just `During the test, TestCafe clicks the link that triggers file download.`

> obtain the name and the content of the file

`obtain the file name and content` - much shorter

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method in combination with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) verify the downloaded file information.
+
+The Request Logger is set up to log all successful `GET` requests to a particular url. The responses are then filtered to only contain a `.txt` file with specified contents.
```
Comment (VasilyStrelyaev) [inline examples/check-downloaded-file-name-and-content/README.MD:11]: URL

**Review by VasilyStrelyaev — CHANGES_REQUESTED** [review]

File: examples/check-downloaded-file-name-and-content/README.MD (hunk 10 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:7]: edited

File: examples/check-downloaded-file-name-and-content/README.MD (hunk 11 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:11]: edited

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 12]
```diff
@@ -0,0 +1,21 @@
+# Check the Name and the Contents of a Downloaded File
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to check the name and the content of a file downloaded during a test.
+
+During the test, TestCafe clicks the link that triggers file download. To verify download success and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
+
+The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method in combination with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) verify the downloaded file information.
```
Comment (VasilyStrelyaev) [inline examples/check-downloaded-file-name-and-content/README.MD:9]: What if we say `name and content`? I don't think file content qualifies as `file information` so we may probably be saying it wrong.

**Review by VasilyStrelyaev — CHANGES_REQUESTED** [review]

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 13]
```diff
@@ -0,0 +1,21 @@
+# Check the Name and the Contents of a Downloaded File
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to check the name and the content of a file downloaded during a test.
+
+During the test, TestCafe clicks the link that triggers file download. To verify download success and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:7]: Sound interchangeable to me. We could use `confirm` instead.

File: examples/check-downloaded-file-name-and-content/README.MD (hunk 12 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:9]: Yes, `name and content` is more precise. I edited it in.

File: examples/check-downloaded-file-name-and-content/README.MD (hunk 13 above)
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply examples/check-downloaded-file-name-and-content/README.MD]: Oh, yes, `confirm` is better

**Review by VasilyStrelyaev — APPROVED** [review]

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 14]
```diff
@@ -0,0 +1,21 @@
+# Check the Downloaded File Name and Content
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to check the name and the content of a file downloaded during a test.
+
+During the test, TestCafe clicks the link that triggers file download. To confirm download success and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
```
Comment (albertov05) [inline examples/check-downloaded-file-name-and-content/README.MD:7]:
```suggestion
During the test, TestCafe clicks the link that triggers the file download. To confirm download completion and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
```

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method in combination with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) verify the downloaded file name and content.
+
+The Request Logger is set up to log all successful `GET` requests to a particular URL. The responses are then filtered to only contain a `.txt` file with specified contents.
```
Comment (albertov05) [inline examples/check-downloaded-file-name-and-content/README.MD:11]:
```suggestion
The Request Logger is set up to log all successful `GET` requests to a particular URL. The responses are then filtered to only contain a `.txt` file with the specified content.
```

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+The Request Logger has the following [options](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/constructor.html) enabled:
+
+|Option|Effect|
+|-|-|
+|`logResponseHeaders`| logs the response header|
+|`logResponseBody`| logs the response body|
+|`stringifyResponseBody`| converts the response body to a string|
+
+The file name is read from the `content-disposition` response header. The file contents constitute the response body.
```
Comment (albertov05) [inline examples/check-downloaded-file-name-and-content/README.MD:21]: `The file contents constitute the response body.`

Can you clarify what you mean by "constitute"? For example, can we say:

`The file contents contain the response body.`

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 17]
```diff
@@ -0,0 +1,21 @@
+# Check the Downloaded File Name and Content
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to check the name and the content of a file downloaded during a test.
+
+During the test, TestCafe clicks the link that triggers file download. To confirm download success and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
+
+The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method in combination with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) verify the downloaded file name and content.
```
Comment (albertov05) [inline examples/check-downloaded-file-name-and-content/README.MD:9]:
```suggestion
The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method, combined with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html), verifies the downloaded file name and content.
```

File: examples/check-downloaded-file-name-and-content/README.MD (hunk 16 above)
Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD:21]: I meant to say that "the response body is the file contents". 
If the response body is literally the only thing inside the file, can we still say `The file contents contain the response body` ?

Comment (albertov05) [reply examples/check-downloaded-file-name-and-content/README.MD]: Thank you. Please change it to:

`The response body contains the contents of the file.`

Comment (DIRECTcut) [reply examples/check-downloaded-file-name-and-content/README.MD]: > The response body is added to the file content.

File: examples/check-downloaded-file-name-and-content/README.MD [hunk 18]
```diff
@@ -0,0 +1,21 @@
+# Check the Downloaded File Name and Content
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to check the name and the content of a file downloaded during a test.
+
+During the test, TestCafe clicks the link that triggers the file download. To confirm download completion and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
+
+The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method, combined with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html), verifies the downloaded file name and content.
```
Comment (albertov05) [inline examples/check-downloaded-file-name-and-content/README.MD:9]:
```suggestion
The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method, combined with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method, verifies the downloaded file name and content.
```

**Pull request merged**